ONSdigital/dp-frontend-filter-dataset-controller
# Job ID is allocated at start of journey to support data persistence

### What

- Add a service to control the filtering of a dataset
- Add 4 HTTP handlers; 1 that creates a job id and 3 that display webpages
- Added CI scripts to the repository
- Includes unit test coverage
- Note that the handlers for "/datasets/{datasetID}/editions/{editionID}/versions/{versionID}" and "/datasets/{datasetID}/editions/{editionID}/versions/{versionID}/filter" should be moved to the dp-frontend-dataset-handler once those repositories have been created

### How to review

- Clone this repository and dp-frontend-renderer (checkout branch feature/job-id)
- Ensure sixteens is running
- Run `make debug` within both of the repositories
- Navigate your browser to http://localhost:20001/datasets/1234/editions/5678/versions/2017
- Follow the user journey by "choosing a filter" and clicking the start button
- This will generate you a job id and redirect you to /jobs/{job_id}
- Finish the journey by clicking on preview and download
- Please note that these are only skeleton templates and do not represent the final user journey
- You should also run `make test` to ensure all tests pass successfully

### Who can review

Anyone, but preferably a frontend dev
